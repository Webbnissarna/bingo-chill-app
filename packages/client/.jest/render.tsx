import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
