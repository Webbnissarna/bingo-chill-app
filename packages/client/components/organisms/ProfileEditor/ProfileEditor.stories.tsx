import type { StoryDefault } from "@ladle/react";
import type { Profile } from ".";
import ProfileEditor from ".";
import { useState } from "react";
import { SPOOKY_GHOST_IMAGE_BASE64 } from "@/.ladle/constants";

export function Default() {
  const [profile, setProfile] = useState<Profile>({
    name: "Me",
    icon: SPOOKY_GHOST_IMAGE_BASE64,
    color: "#8fbcbb",
  });
  return (
    <div className="font-text text-snowStorm-0 bg-polarNight-1 p-5">
      <ProfileEditor
        profile={profile}
        onChanged={(p) => setProfile({ ...p })}
      />
      <span className="text-[8px] font-monospace block w-96 break-all">
        {JSON.stringify(profile, null, 2)}
      </span>
    </div>
  );
}

export default {
  title: "Organisms / Profile Editor",
} satisfies StoryDefault;
