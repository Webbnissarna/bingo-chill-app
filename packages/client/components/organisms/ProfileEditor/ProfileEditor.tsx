import { Base64Icon, Text, TextInputField } from "@/components/atoms";
import ImagePicker from "@/components/molecules/ImagePicker";
import type { DeepPartial } from "@webbnissarna/bingo-chill-common/src/utils/functional";
import { patch } from "@webbnissarna/bingo-chill-common/src/utils/functional";
import { ColorPicker, Modal } from "antd";
import { useState } from "react";

export interface Profile {
  name: string;
  icon: string;
  color: string;
}

export interface ProfileEditorProps {
  profile: Profile;
  onChanged: (newProfile: Profile) => void;
}

export default function ProfileEditor({
  profile,
  onChanged,
}: ProfileEditorProps): JSX.Element {
  const [isModalOpen, setIsModalOpen] = useState(false);

  const onModalCancel = () => {
    setIsModalOpen(false);
  };

  const change = (update: DeepPartial<Profile>) => {
    onChanged(patch(profile, update));
  };

  return (
    <>
      <div
        onClick={() => setIsModalOpen(true)}
        className="w-14 h-14 p-1 rounded-xl bg-polarNight-3 flex items-center justify-center relative cursor-pointer hover:bg-frost-3 active:bg-frost-2 transition"
      >
        <div className="max-h-full aspect-square">
          <Base64Icon src={profile.icon} description="profile" />
        </div>
        <div
          className="w-10/12 h-1 rounded-lg absolute bottom-0"
          style={{ backgroundColor: profile.color }}
        />
      </div>
      <Modal
        title="Edit Profile"
        open={isModalOpen}
        onCancel={onModalCancel}
        footer={[]}
      >
        <div className="w-full grid grid-cols-[1fr_1fr] gap-4 text-right">
          <div className="col-span-2 flex items-center justify-center">
            <ImagePicker
              value={profile.icon}
              onChange={(icon) => change({ icon: icon ?? "" })}
            />
          </div>
          <Text>Name</Text>
          <TextInputField
            value={profile.name}
            onChange={(newName) => change({ name: newName })}
          />
          <Text>Color</Text>
          <div className="text-left">
            <ColorPicker
              disabledAlpha
              value={profile.color}
              onChangeComplete={(value) =>
                change({ color: value.toHexString() })
              }
            />
          </div>
        </div>
      </Modal>
    </>
  );
}
