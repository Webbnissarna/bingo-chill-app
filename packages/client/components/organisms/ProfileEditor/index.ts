export { default } from "./ProfileEditor";
export * from "./ProfileEditor";
