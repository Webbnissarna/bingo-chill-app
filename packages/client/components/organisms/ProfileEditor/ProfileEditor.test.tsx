import { render } from "../../../.jest/render";
import ProfileEditor from ".";

describe("ProfileEditor", () => {
  it("renders", () => {
    const { asFragment } = render(
      <ProfileEditor
        profile={{ name: "", color: "", icon: "" }}
        onChanged={() => undefined}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
