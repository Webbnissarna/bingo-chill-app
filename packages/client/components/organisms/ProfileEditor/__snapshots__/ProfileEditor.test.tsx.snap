// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileEditor renders 1`] = `
<DocumentFragment>
  <div
    class="w-14 h-14 p-1 rounded-xl bg-polarNight-3 flex items-center justify-center relative cursor-pointer hover:bg-frost-3 active:bg-frost-2 transition"
  >
    <div
      class="max-h-full aspect-square"
    >
      <img
        alt="profile"
        class="w-full h-full object-contain"
        src=""
      />
    </div>
    <div
      class="w-10/12 h-1 rounded-lg absolute bottom-0"
    />
  </div>
</DocumentFragment>
`;
